pkg/core/blockchain.go

```diff
@@ -320,7 +320,7 @@ func (bc *Blockchain) AddBlock(block *block.Block) error {
 	}
 	headerLen := bc.headerListLen()
 	if int(block.Index) == headerLen {
-		err := bc.AddHeaders(block.Header())
+		err := bc.addHeaders(false, block.Header())
```

Comment:
Maybe this block should be moved up and called as `bc.addHeaders(bc.config.VerifyBlocks,...)`? Think about it the other way around --- if we already do have a header and it's already verified, why bother with another verification? Only transactions should be verified in this case (if configured to do so).

Reply:
Done.